iridauploader/api/api_calls.py

```diff
@@ -262,6 +262,16 @@ def _get_link(self, target_url, target_key, target_dict=None):
 
         return ret_val
 
+    def _get_upload_url(self, base_url, run_type_str):
+        """
+        Concatenates a base url with the run type for constructing the upload url path
+        :param base_url: Upload url
+        :param run_type_str: Type of sequencing run that is being uploaded
+        :return: url
+        """
+        seq_run_url = self._get_link(base_url, "sequencingRun")
```

Comment:
So it turns out, that we actually have an issue in the IRIDA api that we missed when setting it up to use the `api/sequencingrun/<type>` format. The only relative link that exists is `sequencingRuns/miseq`

The relative link for `sequencingrun` does not exist, in addition, we don't have a wildcard for links like  `sequencingrun/<type>` in the relative links list as far as I can see.

I have changed the code into a working state, and left a detailed comment on how to change it once we have the issue fixed and released on IRIDA.

Reply:
How about this for an alternative solution?  Can we just get the URL from the rel for `sequencingRuns`, then just `urljoin` the sequencer type to the end of that URL?  That way at least we're navigating to the `/sequencingrun` url through HATEOAS, but just appending that sequnecer type because we know its the convention?

Reply:
Technically yes we can, but it's not going to be much cleaner. Since `/sequencingrun` is a finite endpoint, and not a directory endpoint like `/sequencingrun/` (I'm not sure what the actual term for a directory endpoint is). 

I've changed it to work but it kinda goes against HATEOAS no matter which way we do it.